### ISSUE TITLE
How to get the accuracy of a trained model?

### ISSUE THREAD

**Issue by urialon** [typilus/typilus#2]
Hi Miltos,
I wish to reproduce PLDI Table 2 results, without pytype.

I managed to train a model on the PLDI data (without pytype, as we discussed here: #1 ),
create the type map (using `index.py`), and test the model (using `predict.py`) according to the instructions here: [https://github.com/typilus/typilus/tree/master/src](https://github.com/typilus/typilus/tree/master/src)
I also commented out these lines: [https://github.com/typilus/typilus/blob/master/src/typilus/model/codegraphmodel.py#L152-L154](https://github.com/typilus/typilus/blob/master/src/typilus/model/codegraphmodel.py#L152-L154) such that no examples will be dropped (at training, validation and test) (maybe this is incorrect, and I **should** let them be dropped in order for this to be comparable to the PLDI results?).

How can I get the final test accuracy, i.e., the relevant row of Table 2 in the paper?

I saw the instructions here: [https://github.com/typilus/typilus/tree/master/exp/type_check](https://github.com/typilus/typilus/tree/master/exp/type_check)
But I'm not sure which is the relevant command line for me, and whether I should specify `SELECTED_TYPE_CHECKER` or not? Should I use `--prep` or not?

Thanks!

**Comment by mallamanis** [comment]
Hi @urialon 

You can use the `test.py` ([here](https://github.com/typilus/typilus/blob/master/src/typilus/utils/test.py)) to get the results. This outputs a large json with all the metrics and all the metrics per-type. You're interested in [these](https://github.com/typilus/typilus/blob/master/src/typilus/utils/test.py#L138#L139) keys in the dictionary.

**Comment by urialon** [comment]
Thanks!
What should I use for `TYPE_LATTICE_PATH` and `ALIAS_METADATA`?

**Comment by mallamanis** [comment]
During data extraction a `type-lattice.json.gz` file should have been created next to the `.jsonl.gz` graphs, this is the `TYPE_LATTICE_PATH` (see [here](https://github.com/typilus/typilus/blob/9d1cb5db846e8230d0ed8e877021ebf80ea42df6/src/data_preparation/scripts/graph_generator/extract_graphs.py#L125)). For `ALIAS_METADATA` use the path to the `src/data_preparation/metadata/typingRules.json` (in this repo)

**Comment by urialon** [comment]
Sorry for the delay.
I managed to run the `test.py` script.
The weird thing is that I get `"accuracy": 1.0` and `"accuracy_up_to_parametric_type": 1.0` (in the upper-level object).

Here is the output:
[https://gist.github.com/urialon/fa3d6eb8518f944699e9e97834a04c5c](https://gist.github.com/urialon/fa3d6eb8518f944699e9e97834a04c5c)

Both `"count"` and `"count_on_lattice"` are `29092`.
Any idea what am I doing wrong?

**Comment by mallamanis** [comment]
Hi Uri,
Did you maybe index (=build the type map) on the test set (instead of the training/validation set)? This would yield such a behavior...

**Comment by urialon** [comment]
Probably, thanks a lot! I'll re-index
